src/environments/array.js

```diff
@@ -934,6 +959,32 @@ defineEnvironment({
     mathmlBuilder,
 });
 
+defineEnvironment({
+    type: "array",
+    names: ["equation", "equation*"],
+    props: {
+        numArgs: 0,
+    },
+    handler(context) {
+        const settings = context.parser.settings;
+        if (!settings.displayMode) {
+            throw new ParseError(`{${context.envName}} cannot be used inline.`);
+        } else if (settings.strict && !settings.topEnv) {
+            settings.reportNonstrict("textEnv",
+                `{${context.envName}} called from math mode.`);
+        }
```

Comment:
Would it be possible to extract this to a separate function?

Reply:
Done.